### PR TITLE
Move implementation details of che-server workspaces behind services

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Start to move implementation details of che-server workspaces behind services

avoid for example to use `CHE_WORKSPACE_ID` but instead use `WorkspaceService.getCurrentWorkspaceId()`, etc to decouple how che-server is serving those properties

Then it will permit to plug other source of configuration (with DevWorkspace integration for example)

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17901

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

Change-Id: I6c6a6b9d02c3e22dd114dfa53f10456fef3dd0a7
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

**Comment by benoitf** [comment]
[crw-ci-test]

File: extensions/eclipse-che-theia-plugin-ext/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "vscode-uri": "2.1.1",
     "js-yaml": "3.13.1",
     "drivelist": "9.0.2",
+    "@eclipse-che/theia-remote-api": "^0.0.1",
     "@eclipse-che/workspace-telemetry-client": "latest"
   },
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-plugin-ext/package.json:29]: `"@eclipse-che/workspace-client": "latest"` is redundant here?

Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-ext/package.json:29]: hello, I'm not sure to follow your comment ?

Comment (azatsarynnyy) [reply extensions/eclipse-che-theia-plugin-ext/package.json:29]: I mean can we get rid of `workspace-client` dependency in `plugin-ext`?
https://github.com/eclipse/che-theia/blob/6926f0b8c2c4964ff47b1231cbc98ba44e6366f1/extensions/eclipse-che-theia-plugin-ext/package.json#L17
As `theia-remote-api` is going to be used.

Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-ext/package.json:29]: got it, yes sure, will fix in this PR
There will be other PR anyways. Probably splitting remote-api in two (to be sure to cut down transitive dependencies)

Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-ext/package.json:28]: I've removed workspace-client

**Comment by benoitf** [comment]
[crw-ci-test]

**Review by azatsarynnyy — APPROVED** [review]
One step closer to DevWorkspace integration :rocket:

**Comment by benoitf** [comment]
merging, now that 7.20.0 is tagged

**Pull request merged**